casper/src/main/scala/coop/rchain/casper/util/ProtoUtil.scala

```diff
@@ -94,25 +81,43 @@ object ProtoUtil {
     * requires a list to be returned. When we reach the goalFunc,
     * we return an empty list.
     */
-  def getCreatorJustificationAsListUntilGoalInMemory[F[_]: Monad](
+  def getCreatorJustificationAsListUntilGoalInMemory[F[_]: Sync](
```

Comment:
BlockDagRepresentation can have syntax defined in casper project also with this function added.

Reply:
I'd better do that in another PR as the whole ProtoUtil object require refactoring

Reply:
You created already syntax for BlockDagRepresentation in this PR. :wink:

Reply:
I mean not touch ProtoUtils, as lots of that stuff there belong to dag syntax